backend/src/controllers/UserController.js

```diff
@@ -76,9 +82,51 @@ const changePassword = async (req, res) => {
   }
 };
 
+
+const changePFP = async (req, res) => {
+  const userId = req.user.data.user_id;
+  try {
+    upload(req, res, async (err) => {
+      if (err) {
+        console.error('Error uploading file:', err);
+        return res.status(500).send({ error: 'File upload failed' });
+      }
+
+      const file = req.file;
+      if (!file) {
+        return res.status(400).send({ error: 'No file uploaded' });
+      }
+
+      console.log('Uploaded file:', file);
+
+      const buffer = file.buffer;
+      const fileData = buffer.toString('base64');
+      const user = await db.oneOrNone('SELECT * FROM users WHERE user_id = $1', [userId]);
+
+      if (!user) {
+        return res.status(404).send({ error: 'User not found' });
+      }
+
+      const imagePath = `data:image/jpeg;base64,${fileData}`;
+      console.log('Image path:', imagePath);
+
+      const result = await cloudinary.uploader.upload(imagePath, { folder: 'pfps' });
+      console.log('Cloudinary result:', result);
+
+      return res.json({ secure_url: result.secure_url });
+    });
+  } catch (error) {
+    console.error('Error uploading image to Cloudinary:', error.message);
+    return res.status(500).send({ error: 'Internal server error' });
+  }
+};
```

Comment:
- I actually fixed this endpoint on your branch. You probably need to do a git rebase to actually get the write endpoint.
- Instead, i'll just copy the correct endpoint below. 
```suggestion
const changePFP = async (req, res) => {
  const userId = req.user.data.user_id;
  try {
    const file = req.file;
    const buffer = file.buffer;
    const fileData = buffer.toString("base64");
    const defaultPicture =
      "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRXGl68Y0oCfYlx18OswvBI5QNYjr7bHdCCUvAf8lHeig&s";
    const user = await db.oneOrNone("SELECT * FROM users WHERE user_id = $1", [
      userId,
    ]);

    if (user.profileurl !== defaultPicture) {
      const publicID = user.profileurl.split('/').pop().split('.')[0];
      await cloudinary.uploader.destroy('pfps/', publicID);
    }

    const result = await cloudinary.uploader.upload(
      `data:image/jpeg;base64,${fileData}`,
      {
        folder: "pfps",
        transformation: [{ width: 300, height: "300", crop: "fill" }],
      }
    );
    await db.none("UPDATE users SET profileurl = $2 WHERE user_id = $1", [
      userId,
      result.secure_url,
    ]);
    res.status(200).send({
      message: "Profile picture updated",
      profileURL: result.secure_url,
    });
  } catch (error) {
    console.error("Error uploading image to Cloudinary:", error.message);
    return res.status(500).send({ error: "Internal server error" });
  }
};
```